sdk/storage/test/shares/share_client_test.cpp

```diff
@@ -54,4 +66,120 @@ namespace Azure { namespace Storage { namespace Test {
     }
   }
 
+  TEST_F(FileShareClientTest, ShareMetadata)
+  {
+    auto metadata1 = RandomMetadata();
+    auto metadata2 = RandomMetadata();
+    {
+      // Set/Get Metadata works
+      EXPECT_NO_THROW(m_shareClient->SetMetadata(metadata1));
+      auto result = m_shareClient->GetProperties()->Metadata;
+      EXPECT_EQ(metadata1, result);
+      EXPECT_NO_THROW(m_shareClient->SetMetadata(metadata2));
+      result = m_shareClient->GetProperties()->Metadata;
+      EXPECT_EQ(metadata2, result);
+    }
+
+    {
+      // Create share with metadata works
+      auto client1 = Files::Shares::ShareClient::CreateFromConnectionString(
+          AdlsGen2ConnectionString(), LowercaseRandomString());
+      auto client2 = Files::Shares::ShareClient::CreateFromConnectionString(
+          AdlsGen2ConnectionString(), LowercaseRandomString());
+      Files::Shares::CreateShareOptions options1;
+      Files::Shares::CreateShareOptions options2;
+      options1.Metadata = metadata1;
+      options2.Metadata = metadata2;
+
+      EXPECT_NO_THROW(client1.Create(options1));
+      EXPECT_NO_THROW(client2.Create(options2));
+      auto result = client1.GetProperties()->Metadata;
+      EXPECT_EQ(metadata1, result);
+      result = client2.GetProperties()->Metadata;
+      EXPECT_EQ(metadata2, result);
+    }
+  }
+
+  TEST_F(FileShareClientTest, ShareQuota)
+  {
+    const int32_t quota32GB = 32;
+    const int32_t quota64GB = 64;
+    const int32_t quota5120GB = 5120;
+    const int32_t quotaInvalid = 5121;
```

Comment:
if max quota is changed in the future, you'll have to fix this test case.

Reply:
and you're testing server behavior

Reply:
Sure, I'll remove this.